src/main/java/org/mastodon/mamut/tomancak/resolve/CreateConflictTagSet.java

```diff
@@ -0,0 +1,245 @@
+	{
+		if ( index == 0 )
+			return "a";
+		final StringBuilder sb = new StringBuilder();
+		while ( index >= 0 )
+		{
+			sb.append( ( char ) ( 'a' + index % 26 ) );
+			index /= 26;
+			index--;
+		}
+		return sb.reverse().toString();
+	}
+
+	private static RefIntMap< Spot > getBranchIdMap( final Model model )
+	{
+		final ModelBranchGraph branchGraph = new ModelBranchGraph( model.getGraph() );
+		final RefIntMap< Spot > branchIds = new RefIntHashMap<>( model.getGraph().vertices().getRefPool(), -1 );
+		branchGraph.graphRebuilt();
+		final BranchSpot ref = branchGraph.vertexRef();
+		for ( final Spot spot : model.getGraph().vertices() )
+		{
+			final int internalPoolIndex = branchGraph.getBranchVertex( spot, ref ).getInternalPoolIndex();
+			branchIds.put( spot, internalPoolIndex );
+		}
+		return branchIds;
+	}
+
+	private static TIntSet mapToBranchIds( final Set< Spot > conflict, final RefIntMap< Spot > branchIds )
+	{
+		final TIntSet branchIdsInConflict = new TIntHashSet();
+		for ( final Spot spot : conflict )
+			branchIdsInConflict.add( branchIds.get( spot ) );
+		return branchIdsInConflict;
+	}
+
+	/**
+	 * For the given frame, return a list of sets of spots that are in conflict.
+	 * Two spots are in conflict if their Hellinger distance is less than the threshold.
+	 */
+	private static List< Set< Spot > > findConflictsForFrame( final ModelGraph graph, final SpatialIndex< Spot > frame, final double threshold )
+	{
+		// This method returns the connected components of the conflict graph.
+		// Reimplementing this method with a proper connected components algorithm would be a good idea.
+		final IncrementalNearestNeighborSearch< Spot > nearestNeighbors = frame.getIncrementalNearestNeighborSearch();
+		final RefObjectMap< Spot, Set< Spot > > conflicts = new RefObjectHashMap<>( graph.vertices().getRefPool() );
+		for ( final Spot spot : frame )
+		{
+			final RefSet< Spot > group = findConflictsForSpot( graph, spot, nearestNeighbors, threshold );
+			if ( group != null && !group.isEmpty() )
+			{
+				group.add( spot );
+				mergeGroups( group, conflicts );
+			}
+		}
+		return new ArrayList<>( conflicts.values() );
+	}
+
+	/**
+	 * Return a set of spots that the given spot is in conflict with.
+	 * Two spots are in conflict if their Hellinger distance is less than the threshold.
+	 * <p>
+	 * For efficiency, we only check the 10 nearest neighbors.
+	 */
+	private static RefSet< Spot > findConflictsForSpot( final ModelGraph graph, final Spot spot, final IncrementalNearestNeighborSearch< Spot > nearestNeighbors, final double threshold )
+	{
+		// NB: Having a fixed number of neighbors to check is a bad idea.
+		//    This is a quick hack to make the method run in reasonable time.
+		//    It would be better to derive a distance threshold from the Hellinger
+		//    distance equation, Hellinger distance threshold and the spot radius.
+		RefSet< Spot > group = null;
+		nearestNeighbors.search( spot );
+		for ( int i = 0; i < 10; i++ )
```

Comment:
You could consider to use `while(nearestNeighbors.hasNext())`

This would feel a bit more intuitive way how to use the iterator. The number of 10 could be used as a break condition for the while-loop.